src/libs/Violations/ViolationsUtils.ts

```diff
@@ -78,6 +79,53 @@ const ViolationsUtils = {
                 if (!hasMissingTagViolation && !updatedTransaction.tag && policyRequiresTags) {
                     newTransactionViolations.push({name: CONST.VIOLATIONS.MISSING_TAG, type: 'violation'});
                 }
+            } else {
```

Comment:
NAB: can we consider extracting some part of this new code into a helper function? I find the logic hard to follow with all the if/else/for statements nested together. Abstracting out some of the logic would make this a lot more readable IMO.